File: 230110-dojo/java/app/src/main/java/Solution.java
class Solution {
    public int solution(int left, int right) {
        int accumulator = 0;

        for (int i = left; i <= right; i += 1) {
            int count = 0;

            for (int j = 1; j <= i; j += 1) {
                count += i % j == 0 ? 1 : 0;
            }

            accumulator += count % 2 == 0 ? i : i * (- 1);
        }

        return accumulator;
    }
}

File: 230110-dojo/javascript/solution.js
export function makeArray(left, right) {
  return Array(right - left + 1).fill(0).map((_, index) => left + index);
}

export default function solution(left, right) {
  const array = makeArray(left, right);

  const resultArray = array.map((value) => {
    let count = 0;

    for (let i = 1; i <= value; i += 1) {
      count += value % i === 0 ? 1 : 0;
    }

    return count % 2 === 0 ? value : value * (-1);
  });

  return resultArray.reduce((pre, cur) => pre + cur, 0);
}
